### PR TITLE
Create a custom level editor as a second tab within the level editor tab.

### PR DESCRIPTION
**Pull request by jaythebusinessgoose**
The custom level editor is for editing level files which represent a full static level. Each template in these level files is a setroom template with the coordinates the room should appear at; any other room is ignored.

This editor does not show any level files with the same name as a vanilla level file so that those aren't accidentally modified in a way that could break them.

There have also been a few fixes for the regular editor:
- Save file icons are updated upon saving.
- Fixed scroll wheel in the "Full Level View".

### CODE REVIEW

**Comment by gmjosack** [comment]
Looks like you've added a ton of new instance variables defined outside of `__init__` which is an anti-pattern and causing pylint to fail

**Comment by jaythebusinessgoose** [comment]
Most of those were pre-existing so I was kinda confused. I guess I can just add all of them to the __init__.

I wasn't sure if there was a reason they were left out, but if it's an anti-pattern causing the linter to fail, that's a good enough reason for me to add them.

**Comment by jaythebusinessgoose** [comment]
Resolved those issues.

File: src/modlunky2/config.py [hunk 1]
```diff
@@ -164,6 +167,16 @@ def from_path(cls, config_path: Path, exe_dir=None):
...
 
+        # Level Editor
+        obj.level_editor_tab = config_data.get("level_editor_tab")
+
+        obj.custom_level_editor_custom_save_formats = config_data.get(
+            "custom_level_editor_custom_save_formats"
+        )
+        obj.custom_level_editor_default_save_format = config_data.get(
+            "custom_level_editor_default_save_format"
+        )
```
Comment (gmjosack) [inline src/modlunky2/config.py:178]: keys in the config are defined with `kebab-case`, not `snake_case`

File: src/modlunky2/ui/levels.py [hunk 2]
```diff
@@ -31,13 +33,52 @@
...
+
+
+class CustomLevelSaveFormat:
+    def __init__(self, name, room_template_format, include_vanilla_setrooms):
+        self.name = name or room_template_format
+        self.room_template_format = room_template_format
+        self.include_vanilla_setrooms = include_vanilla_setrooms
+
+    @classmethod
+    def LevelSequence(cls):
```
Comment (gmjosack) [inline src/modlunky2/ui/levels.py:54]: python functions and methods use `snake_case` not `CamelCase`

**Review by gmjosack — COMMENTED** [review]
nits for now

File: src/modlunky2/ui/levels.py (hunk 2 above)
Comment (jaythebusinessgoose) [reply src/modlunky2/ui/levels.py]: Done.

File: src/modlunky2/config.py (hunk 1 above)
Comment (jaythebusinessgoose) [reply src/modlunky2/config.py]: Done.

**Pull request merged**